### ISSUE TITLE
Core Animation rendering engine doesn't support gradients with complex alpha value locations

### ISSUE THREAD

**Issue by JoeSzymanski** [airbnb/lottie-ios#1636]
We have started looking into animations we've previously had performance issues with in Lottie 3.x, including ones related to the fork we created at https://github.com/themeetgroup/lottie-ios

One of the first animations we have tested has shown issues in Automatic and CoreAnimation modes, seemingly related to alpha.

Here is the animation in our existing version of Lottie and when using the MainThread mode:

Legacy Lottie version:
https://user-images.githubusercontent.com/1273497/177864781-7842c893-2194-4565-8001-401e1a41eb91.mov
MainThread mode:
https://user-images.githubusercontent.com/1273497/177864845-f1d3318b-60dc-4c22-ae19-19afd40b395e.mov

Here is the animation in Automatic and CoreAnimation modes:
Automatic:
https://user-images.githubusercontent.com/1273497/177864912-d3565879-5aab-4a63-8a26-64e99dd97f96.mov
CoreAnimation:
https://user-images.githubusercontent.com/1273497/177864937-a0f0782f-f4f9-4f1d-977d-9c36b69d5dc9.mov

The animation file is attached.
[zodiac-sagittarius.json.zip](https://github.com/airbnb/lottie-ios/files/9067099/zodiac-sagittarius.json.zip)

**Comment by calda** [comment]
Testing on master, I think this was partially fixed by #1634 -- the gradient around the individual stars look correct now when using the Core Animation rendering engine, but the gradient that covers the rest of the animation isn't quite correct. 

`RenderingEngineOption.automatic` behaves correctly and falls back to the main thread rendering engine (since this animation isn't displayed correctly in the Core Animation engine at this time), outputting:

```
Encountered Core Animation compatibility issue while setting up animation:
[animation.background.Rectangle 1.Gradient Fill 1] The Core Animation rendering engine currently expects gradient colors and alpha values to be in the same order, with the same color stops / locations.
This animation may have additional compatibility issues, but animation setup was cancelled early to avoid wasted work.
```

To support this in the Core Animation rendering engine, we'll need to update how it renders gradients with alphas to support this specific configuration.

| Automatic (falls back to main thread engine) | Core animation engine |
| --- | --- |
| ![Simulator Screen Shot - iPhone 8 - 2022-07-07 at 13 28 12](https://user-images.githubusercontent.com/1811727/177866200-e406d3ee-6aa3-4065-9b16-8f9618a6b513.png) | ![Simulator Screen Shot - iPhone 8 - 2022-07-07 at 13 27 58](https://user-images.githubusercontent.com/1811727/177866215-21032717-1a2d-4a43-bb10-c3d116c67223.png) |

**Comment by JoeSzymanski** [comment]
I will check again against master, but in the 3.4.0 release, Automatic mode was not falling back to MainThread mode for this animation.

**Comment by calda** [comment]
That bug specifically was fixed by #1634

**Comment by calda** [comment]
Fixed in #1640 -- the Core Animation rendering engine now supports gradients with different color stops / locations for the RGB color components and alpha components.

**Comment by JoeSzymanski** [comment]
Thank you for the quick fix. We will continue looking into our large set of animations to see if there are any others that pose issues with the new modes.
